### PR TITLE
SALTO-6317: Move Jsm forms to official api

### PR DESCRIPTION
**Pull request by idoamit8**
In this PR I moved the Form type in Jira to the new API 

---

_Additional context for reviewer_
The values ​​remained the same except for 3 new fields which I will add Noise Reduction.

image

  
From what I see through the CLI - fetch, create, modify and remove all worked great! 
It is important to note some point:
1. The id changed from a number to a string which is actually a uuid and I'm not sure if it has certain meanings or not
since the uuid was deleted, so the ServiceID changed to the id field. Whats going to happen in the next Fetch? 

---
_Release Notes_: 
_Jira Adapter_:
* Moved JSM forms to the new API. 

---
_User Notifications_: 
_Jira Adapter:_
* Changes will be applied in Form instances:
The fields `issueRequestTypeIds`, `newIssueIssueTypeIds`, `newIssueRequestTypeIds` will be replaced by  `recommendedIssueRequestTypeIds`, `issueCreateIssueTypeIds`, `issueCreateRequestTypeIds`.

### CODE REVIEW

File: packages/jira-adapter/src/client/client.ts [hunk 1]
```diff
@@ -211,6 +211,17 @@ export default class JiraClient extends clientUtils.AdapterHTTPClient<Credential
...
 
+  public async atlassianApiSendRequest(
+    method: keyof clientUtils.HttpMethodToClientParams,
+    args: clientUtils.ClientDataParams,
+  ): Promise<clientUtils.Response<clientUtils.ResponseValue | clientUtils.ResponseValue[]>> {
+    const cloudId = await this.getCloudId()
+    return this.sendRequest(method, {
+      ...args,
+      url: `https://api.atlassian.com/jira/forms/cloud/${cloudId}/${args.url}`,
+    })
```
Comment (IdoZakk) [inline packages/jira-adapter/src/client/client.ts:222]: When you use sendRequest you miss the decorators of the rate limit. 
We should have a call for each of the REST calls (you can add only the ones you currently use)

File: packages/jira-adapter/test/filters/forms/forms.test.ts [hunk 2]
```diff
@@ -693,123 +658,33 @@ describe('forms filter', () => {
...
       elements = [projectInstance, projectType, formPortalType, formPublishType, formType, requestTypeType]
     })
     it('should add form', async () => {
       const res = await filter.deploy([{ action: 'add', data: { after: formInstance } }])
       expect(res.leftoverChanges).toHaveLength(0)
       expect(res.deployResult.errors).toHaveLength(0)
       expect(res.deployResult.appliedChanges).toHaveLength(1)
-      expect(connection.post).toHaveBeenCalledTimes(2)
-      expect(connection.put).toHaveBeenCalledTimes(1)
-      expect(connection.put).toHaveBeenCalledWith(
```
Comment (IdoZakk) [inline packages/jira-adapter/test/filters/forms/forms.test.ts:729]: Maybe put the equivalent expects here?

File: packages/jira-adapter/test/filters/forms/forms.test.ts [hunk 3]
```diff
@@ -818,16 +693,12 @@ describe('forms filter', () => {
       expect(res.leftoverChanges).toHaveLength(0)
       expect(res.deployResult.errors).toHaveLength(0)
       expect(res.deployResult.appliedChanges).toHaveLength(1)
-      expect(connection.put).toHaveBeenCalledTimes(1)
-      expect(connection.post).toHaveBeenCalledTimes(1)
```
Comment (IdoZakk) [inline packages/jira-adapter/test/filters/forms/forms.test.ts:822]: I see you removed those in all the tests, I think we should verify the outside calls are called

**Review by IdoZakk — COMMENTED** [review]
🆕 🥇 
Very exciting!
You mentioned the change of number to uuid. I do not think there is anything that might happen here, but we should notify the CEs that all deploys must be done after a new fetch
Also you mentioned a few fields changing, I did not see the removal of the templateId (which was removed in the code

File: packages/jira-adapter/src/client/client.ts (hunk 1 above)
Comment (idoamit8) [reply packages/jira-adapter/src/client/client.ts:222]: I use all of them. I will change it to use each call

**Comment by idoamit8** [comment]
@IdoZakk I resolved your comments :) 
What do you mean by 

> you mentioned a few fields changing, I did not see the removal of the templateId (which was removed in the code

Where did you see it?

**Comment by IdoZakk** [comment]
> @IdoZakk I resolved your comments :) What do you mean by
> 
> > you mentioned a few fields changing, I did not see the removal of the templateId (which was removed in the code
> 
> Where did you see it?

In the PR comment, under `Additional context for reviewer`

**Review by IdoZakk — APPROVED** [review]
💣

**Pull request merged**